File: 백준/Silver/1966. 프린터 큐/프린터 큐.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');

let numTestCases = Number(input.shift());

for (let i = 0; i < numTestCases; i++) {
    const [N, M] = input.shift().split(' ').map(Number);
    const priorities = input.shift().split(' ').map(Number);
    
    let queue = priorities.map((priority, index) => {
        return { priority, index };
    });

    let count = 0;
    while (queue.length > 0) {
        const maxP = Math.max(...queue.map(x => x.priority));
        const current = queue.shift();
        if (current.priority < maxP) {
            queue.push(current);
        } else {
            count++;
            if (current.index === M) {
                console.log(count);
                break;
            }
        }
    }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
from collections import deque

T = int(input())

for _ in range(T):
    N, M = map(int, input().split())
    priorities = list(map(int, input().split()))
    queue = deque([(p, i) for i, p in enumerate(priorities)])
    
    count = 0
    while queue:
        max_p = max(queue, key=lambda x: x[0])[0]
        
        current = queue.popleft()
        
        if current[0] < max_p:
            queue.append(current)
        else:
            count += 1
            if current[1] == M:
                print(count)
                break